### ISSUE TITLE
Pod status out of sync after being marked as not ready by controller manager

### ISSUE THREAD

**Issue by turkenh** [virtual-kubelet/virtual-kubelet#899]
---

### Environment summary

Provider: Custom provider for Kubernetes

Version: v1.3.0

K8s Master Info: GKE

Install Method: Raw manifest

### Issue Details

After pods created, if pod status in k8s updated by another entity (e.g. controller manager), status never gets synced from provider, because of this check: https://github.com/virtual-kubelet/virtual-kubelet/pull/824

### Repro Steps

1. Create pod in k8s
2. Provider creates the pod and syncs its status back
3. Pod in k8s ready/running, all fine
4. Virtual kubelet fails to update node status for some time for whatever reason (e.g. network connectivity issues)
5. Virtual node marked as `NotReady` with message: `Kubelet stopped posting node status`
6. kube-controller-manager of k8s, goes and marks all pods as Ready = false:

```
I1105 12:40:27.046070       1 controller_utils.go:139] Updating ready status of pod busybox-545fd8cc6c-zmwdb to false
```

7. Virtual kubelet never sync's status of pod in provider back to k8s because of [this check ](https://github.com/virtual-kubelet/virtual-kubelet/pull/824), hence pod stuck as `NotReady`.

Please note, this is just one possible case, there could be other cases where status of pod in k8s could be updated by another entity than provider/virtual kubelet.

/cc @cwdsuzhou

**Comment by sargun** [comment]
@cwdsuzhou I suggest we completely remove the check, or we add an additional check when we get new pods from K8s to see if our pod status does not match theirs and if we skipped an update, and if so, trigger re-update on our side.

**Comment by cwdsuzhou** [comment]
IMO we should not remove this check. Maybe we should trigger re-renque of pod in provider if pod status in K8s change rather than remove this change. If remove this change, we would get more and more requests of pod status update.

**Comment by sargun** [comment]
@cwdsuzhou Did you see https://github.com/virtual-kubelet/virtual-kubelet/pull/902?

**Comment by turkenh** [comment]
I think this is fixed now with #902
